service/src/java/org/apache/hive/service/cli/operation/SQLOperation.java

```diff
@@ -319,11 +319,7 @@ public void run() {
         @Override
         public Object run() throws HiveSQLException {
           assert (!parentHive.allowClose());
-          try {
-            Hive.set(parentSessionState.getHiveDb());
-          } catch (HiveException e) {
-            throw new HiveSQLException(e);
-          }
+          Hive.set(parentHive);
```

Comment:
does it completely revert HIVE-25085?

Reply:
Thank you for the comment!

It doesn't revert [HIVE-25085](https://issues.apache.org/jira/browse/HIVE-25085), the `parentHive` is got from the parent session, where ensures that `parentHive` wouldn't be shared across the session.